File: 900/1373B/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;
    
    while (t--) 
    {
        string s;
        cin >> s;

        int a = 0; // count of '0'
        for (char ch : s) 
        {
            if (ch == '0') 
                a++;
        }

        int b = (int)s.size() - a; // count of '1'

        if (a == b && (a % 2 == 0))
            cout << "NET\n";
        else if (a == b && (a % 2 == 1))
            cout << "DA\n";
        else 
        {
            if (min(a, b) % 2 == 0)
                cout << "NET\n";
            else
                cout << "DA\n";
        }
    }

    return 0;
}

File: 900/1373B/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt();
        while(t-- > 0)
        {
            String s = sc.next();
            
            int a = 0;
            
            for(int i=0;i<s.length();i++)
            {
                if(s.charAt(i) == '0')
                    a++;
            }
            
            int b = s.length()-a;
            
            if(a == b && ((a&1) == 0))
                System.out.println("NET");
            else if(a == b && ((a&1) == 1))
                System.out.println("DA");
            else
            {
                if((Math.min(a,b) & 1) == 0)
                    System.out.println("NET");
                else
                    System.out.println("DA");
            }
        }
    }
}